### PR TITLE
fix(payments-next): Move searchParams to params

### PR DESCRIPTION
**Pull request by xlisachan**
## Because

- To access certain path parameters from `layout.tsx`

## This pull request

- [x] Moves the following searchParams to path params and update relevant pages.
   - [x] locale
   - [x] interval
 
Current URL
`https://subscriptions.firefox.com/{offering_identifier}/{intent}/{cart_id}/{page_name}?interval=&promotion_code=&experiment=&locale=`

Proposed URL
`https://subscriptions.firefox.com/{locale}/{offering_identifier}/{intent}/{interval}/{cart_id}/{page_name}?promotion_code=&experiment=`

## Issue that this pull request solves

Closes: FXA-9367

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.

### CODE REVIEW

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/success/page.tsx [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
+import {
+  getCartData,
+  getContentfulContent,
+} from '../../../../../../_lib/apiClient';
 import { headers } from 'next/headers';
 import Image from 'next/image';
 import checkLogo from '../../../../../images/check.svg';
 import circledConfirm from '../../../../../images/circled-confirm.svg';
-import { formatPlanPricing } from '../../../../../../../../libs/payments/ui/src/lib/utils/helpers';
+import { formatPlanPricing } from '../../../../../../../../../../libs/payments/ui/src/lib/utils/helpers';
```
Comment (StaberindeZA) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/success/page.tsx:10]: Could this be updated to use the TS alias instead?

Comment (xlisachan) [reply apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/success/page.tsx:10]: I hope so, that is just awful!

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/error/page.tsx
Comment (pdehaan) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/error/page.tsx:1]: Unrelated nit, but&hellip; do we have a policy [yet] on copyright headers in new files?

```sh
git grep -PLin "http://mozilla.org/MPL/2.0/" -- apps/**/*.{js,jsx,ts,tsx} | cat

apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/error/page.tsx
apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/page.tsx
apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/success/page.tsx
apps/payments/next/app/[locale]/[offeringId]/checkout/layout.tsx
apps/payments/next/app/[locale]/[offeringId]/checkout/new/page.tsx
apps/payments/next/app/_lib/apiClient.ts
apps/payments/next/app/_lib/stubs.ts
apps/payments/next/app/api/auth/[...nextauth]/route.ts
apps/payments/next/app/layout.tsx
apps/payments/next/app/page.tsx
apps/payments/next/auth.config.ts
apps/payments/next/auth.ts
apps/payments/next/index.d.ts
apps/payments/next/jest.config.ts
apps/payments/next/next-env.d.ts
apps/payments/next/next.config.js
apps/payments/next/postcss.config.js
apps/payments/next/tailwind.config.ts
apps/payments/next/tests/app/page.test.tsx
```

Comment (xlisachan) [reply apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/error/page.tsx:1]: I'll add them in! Thanks @pdehaan!

**Review by StaberindeZA — APPROVED** [review]
lgtm. Thank you for the additional changes

**Pull request merged**